lib/time.ml

```diff
@@ -50,14 +50,20 @@ type sleep = {
   thread : unit Lwt.u;
 }
 
+(** bheap library only operates on maximums, but we need the minimum,
+    so invert the order *)
+let invert_comparison x = -x
```

Comment:
I'd move this inside of the `Binary_heap.Make` into the `compare` there. IMHO no need to have this binding at top-level.